util/chplenv/chpl_gpu.py

```diff
@@ -98,7 +98,7 @@ def get_arch():
     gpu_type = get()
 
     # No arch if GPU is not being used.
-    if gpu_type == 'none':
+    if gpu_type == 'none' or gpu_type == 'cpu':
```

Comment:
I suspect it's not possible for `gpu_type` to be `none` but I don't see an issue with this "as is" (if we did have a 'none' type having a 'none' arch would seem to be the sensible choice).

Reply:
We do have the following:
```Python
@memoize
def get():
    if chpl_locale_model.get() != 'gpu':
        return 'none'
```

Which means that it can be none if the locale model is not GPUs. On the other hand, `compute_internal_values` always goes for calling `get_arch`, even if the GPU locale model is `none`. So I think this _can_ happen.

Reply:
I was able to trigger this error by switching away from the `gpu` locale if I removed the `none` check.

```
➜  chapel git:(fix-gpu-cpu-arch) ✗ ./util/printchplenv --all --internal
Traceback (most recent call last):
  File "/Users/daniel.fedorin/Documents/software/chapel/util/chplenv/printchplenv.py", line 550, in <module>
    main()
  File "/Users/daniel.fedorin/Documents/software/chapel/util/chplenv/printchplenv.py", line 543, in main
    compute_internal_values()
  File "/Users/daniel.fedorin/Documents/software/chapel/util/chplenv/printchplenv.py", line 300, in compute_internal_values
    ENV_VALS['  CHPL_GPU_ARCH'] = chpl_gpu.get_arch()
                                  ^^^^^^^^^^^^^^^^^^^
  File "/Users/daniel.fedorin/Documents/software/chapel/util/chplenv/utils.py", line 44, in memoize_wrapper
    cache[args] = func(*args)
                  ^^^^^^^^^^^
  File "/Users/daniel.fedorin/Documents/software/chapel/util/chplenv/chpl_gpu.py", line 110, in get_arch
    arch = GPU_TYPES[gpu_type].default_arch
           ~~~~~~~~~^^^^^^^^^^
KeyError: 'none'
```